palace/fem/lumpedelement.cpp

```diff
@@ -12,90 +12,70 @@
     Mpi::Warning("User specified direction {} does not align with either bounding box "
                  "axis up to {:.3e} degrees!\n"
                  "Axis 1: {} ({:.3e} degrees)\nAxis 2: {} ({:.3e} degrees)\nAxis 3: "
                  "{} ({:.3e} degrees)!\n",
-                 input_dir, angle_warning_deg, normal_0, deviation_deg[0], normal_1,
-                 deviation_deg[1], normal_2, deviation_deg[2]);
+                 input_dir, angle_warning_deg, normals[0], deviations_deg[0], normals[1],
+                 deviations_deg[1], normals[2], deviations_deg[2]);
   }
-  MFEM_VERIFY(std::any_of(deviation_deg.begin(), deviation_deg.end(),
+  MFEM_VERIFY(std::any_of(deviations_deg.begin(), deviations_deg.end(),
                           [](double x) { return x < angle_error_deg; }),
               "Specified direction does not align sufficiently with bounding box axes ("
-                  << deviation_deg[0] << ", " << deviation_deg[1] << ", "
-                  << deviation_deg[2] << " vs. tolerance " << angle_error_deg << ")!");
+                  << deviations_deg[0] << ", " << deviations_deg[1] << ", "
+                  << deviations_deg[2] << " vs. tolerance " << angle_error_deg << ")!");
   direction.SetSize(input_dir.size());
   std::copy(input_dir.begin(), input_dir.end(), direction.begin());
   direction /= direction.Norml2();
 
   // Compute the length from the most aligned normal direction.
-  l = lengths[std::distance(deviation_deg.begin(),
-                            std::min_element(deviation_deg.begin(), deviation_deg.end()))];
-  MFEM_ASSERT((l - mesh::GetProjectedLength(mesh, attr_marker, true, input_dir)) / l <
-                  rel_tol,
+  constexpr double rel_tol = 1.0e-6;
+  auto l_component =
+      std::distance(deviations_deg.begin(),
+                    std::min_element(deviations_deg.begin(), deviations_deg.end()));
+  l = lengths[l_component];
+  MFEM_VERIFY(std::abs(l - mesh::GetProjectedLength(mesh, attr_marker, true, input_dir)) <
+                  rel_tol * l,
               "Bounding box discovered length should match projected length!");
-  w = A / l;
+
+  // Compute the width of the rectangular lumped element.
+  lengths[l_component] = 0.0;
+  if (bounding_box.planar)
+  {
+    // Compute the width from the largest dimension excluding the length direction
+    // component.
+    auto w_component =
+        std::distance(lengths.begin(), std::max_element(lengths.begin(), lengths.end()));
+    MFEM_VERIFY(
+        w_component != l_component,
+        "Bounding box length and width of should correspond to different components!");
+    w = lengths[w_component];
+  }
+  else
+  {
+    // For a non-planar element (shell of a rectangular prism, which can be "unfolded" to a
+    // planar rectangle), compute the width as the perimeter of the bounding box orthogonal
+    // to the length direction.
+    w = 2.0 * std::accumulate(lengths.begin(), lengths.end(), 0.0);
```

Comment:
My understanding is the geometry in https://github.com/awslabs/palace/issues/89 was not L-shaped but was the entire shell of the rectangular prism (4 quad surfaces, not 2). But I do suppose an L-shaped port could also be valid and indeed would not be addressed by this. Maybe we have to fall back to the area calculation but that seems like a weird use case.

Reply:
From what I remember working on that issue (it was a while ago), the geometry was that L shape, which was what threw me off in debugging it, as I had assumed everything would be planar for lumped. That was the reason I stuck with the area calculation as opposed to pulling directly off the bounding box, then only if planar checked that it was a rectangle.

Reply:
Yep. To address this concern and also support #89, we'll take the following approach:
- Add a check for `bounding_box.planar`, like we do for coaxial lumped elements, and remove the second branch if the if/else which makes this 4-sided assumption.
- For the case of #89, each side can be a separate element for a multielement lumped port, as:
  ```
  "LumpedPort": [{
    "Index": 1,
    "Elements": [{
      "Attributes": [<attribute 1>],
      "Direction": "-Y"
    },
    {
      "Attributes": [<attribute 2>],
      "Direction": "+Y"
    },
    ...
  ]
  ```

This will give the correct behavior for the impedance BC, with each planar rectangular element adding in parallel much like a usual multielement lumped port.

I've added this in [e4366f7](https://github.com/awslabs/palace/pull/248/commits/e4366f7af40c29d57fa8a17cad3510ec02252637) and claified the requirement in the docs in 
8d24299.